compressor/templatetags/compress.py

```diff
@@ -90,36 +90,48 @@ def render_cached(self, compressor, kind, mode):
         if self.is_offline_compression_enabled(forced) and not forced:
-            return self.render_offline(context)
-
+            compressed_content = self.render_offline(context)
         # Take a shortcut if we really don't have anything to do
-        if (not settings.COMPRESS_ENABLED and
+        elif (not settings.COMPRESS_ENABLED and
                 not settings.COMPRESS_PRECOMPILERS and not forced):
-            return self.get_original_content(context)
-
-        name = name or getattr(self, 'name', None)
-        context['compressed'] = {'name': name}
-        compressor = self.get_compressor(context, kind)
-
-        # Check cache
-        cache_key = None
-        if settings.COMPRESS_ENABLED and not forced:
-            cache_key, cache_content = self.render_cached(compressor, kind, mode)
-            if cache_content is not None:
-                return cache_content
-
-        file_basename = name or getattr(self, 'basename', None)
-        if file_basename is None:
-            file_basename = 'output'
-
-        rendered_output = compressor.output(mode, forced=forced, basename=file_basename)
-        assert isinstance(rendered_output, six.string_types)
-        if cache_key:
-            cache_set(cache_key, rendered_output)
-        return rendered_output
+            compressed_content = self.get_original_content(context)
+        else:
+            name = name or getattr(self, 'name', None)
+            context['compressed'] = {'name': name}
+            compressor = self.get_compressor(context, kind)
+
+            # Check cache
+            cache_key = None
+            if settings.COMPRESS_ENABLED and not forced:
+                cache_key, cache_content = self.render_cached(compressor, kind, mode)
+                if cache_content is not None:
+                    compressed_content = cache_content
+            else:
+                file_basename = name or getattr(self, 'basename', None)
+                if file_basename is None:
+                    file_basename = 'output'
+
+                rendered_output = compressor.output(mode, forced=forced, basename=file_basename)
+                assert isinstance(rendered_output, six.string_types)
+                if cache_key:
+                    cache_set(cache_key, rendered_output)
+                compressed_content = rendered_output
+
+        if compressed_content is None:
+            file_basename = name or getattr(self, 'basename', None)
+            if file_basename is None:
+                file_basename = 'output'
+            compressed_content = compressor.output(mode, forced=forced, basename=file_basename)
+        # Do the CSP nonce replacement
+        if context.get('django_csp_nonce'):
+            compressed_content = compressed_content.replace('script ', 'script nonce={{request.nonce}} ')
+            compressed_content = compressed_content.replace('type="text/css"', 'type="text/css" nonce={{request.nonce}} ')
```

Comment:
Sorry, these 2 `replace` calls are super dodgy. Why didn't you follow my suggestion to add a specific string we could look for in the template and search&replace *that* instead?

More importantly, I don't see how that helps. You now are outputting HTML that contains `nonce={{request.nonce}}`, which is then not parsed by the template engine, so the litteral nonce you get is `{{request.nonce}}` and not the actual nonce value.

This is why I suggested to replace directly with the `context['request'].csp_nonce` value.

Reply:
I agree they are dodgy, I was just trying to get something to pass testing and I thought this might be equivalent. I suppose not. I've push an update to the PR which reintroduces a tag and looks for that as a replacement. I was unaware that the output would not be revisited by the django csp plugin. 

I'm still unsure of how to force the if case in the template to be true during testing.